icu4c/source/test/intltest/regcoll.cpp

```diff
@@ -1249,6 +1249,17 @@ void CollationRegressionTest::TestBeforeWithTooStrongAfter() {
     }
 }
 
+void CollationRegressionTest::TestICU22517() {
+    IcuTestErrorCode errorCode(*this, "TestICU22517");
+    char16_t data[] = u"&a=b쫊쫊쫊쫊쫊쫊쫊쫊";
+    icu::UnicodeString rule(true, data, -1);
+    for (int i = 4; i <= rule.length(); i++) {
```

Comment:
How long does this test run? Should we test only up to rule.length() - 1 or - 2 unless in exhaustive mode?

Reply:
55.5s

Reply:
Almost a minute... is too long I think for normal testing.
Please test only up to rule.length() - 1 or - 2 unless in exhaustive mode.